platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/OperationRunner.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.util;
+
+import java.util.function.Consumer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The OperationRunner will execute an {@link Operation} and re-execute it if an exception is
+ * thrown. This class only has static methods; the class cannot be instantiated.
+ */
+public class OperationRunner {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OperationRunner.class);
+
+    private static final long SLEEP_FACTOR = 2;
+
+    private OperationRunner() {
+        // prevent instances
+    }
+
+    /**
+     * This method will execute an {@link Operation} and retry if it throws an exception. If an
+     * exception is thrown and retry attempts are still remaining, then {@code exceptionHandler}
+     * will be called and the method will sleep. The sleep duration is doubled after each retry
+     * attempt and is limited to {@code maxSleep}. If there are no retry attempts remaining, then
+     * the exception will be rethrown.
+     * <p>
+     * The caller should be careful when selecting values for {@code maxAttempts},
+     * {@code initialSleep} and {@code maxSleep} because this method could theoretically
+     * run forever.
+     *
+     * @param maxAttempts      the maximum retry attempts
+     * @param initialSleep     the number of milliseconds to sleep after the first failed attempt, limited to maxSleep
+     * @param maxSleep         the maximum number of milliseconds to sleep after any failed attempt
+     * @param operation        the operation to be executed
+     * @param exceptionHandler handle exceptions between retry attempts
+     * @throws Exception the final exception thrown by the operation
+     */
+    public static void runWithRetry(int maxAttempts, long initialSleep, long maxSleep,
+            Operation operation, Consumer<Exception> exceptionHandler) throws Exception {
+
+        int attempt = 1;
+        long sleep = Math.min(initialSleep, maxSleep);
+
+        while (true) {
+            try {
+                operation.run();
+                return;
+            } catch (Exception e) {
```

Comment:
I don't think catching Error is right in this situation. The intended use of this utility to execute something that should succeed, but may require a retry due to timing or network issues. I wouldn't want to catch things like NoSuchMethod because I wouldn't expect that to quickly resolve itself. Also, I don't foresee this utility being used to execute like a container or plugin manager. However, I will remain open to discussing it further if you wish.